corehq/apps/couch_sql_migration/casepatch.py

```diff
@@ -245,6 +263,52 @@ def add_patch_operation(sql_form):
     ))
 
 
+def get_diff_block(case):
+    """Get XML element containing case diff data
+
+    Some early patch forms were submitted without this element.
+
+    :param case: `PatchCase` instance.
+    :returns: A "<diff>" XML element string containing XML-escaped
+    JSON-encoded case diff data, some of which may be patched.
+
+    ```json
+    {
+        "case_id": case.case_id,
+        "diffs": [
+            {
+                "path": diff.path,
+                "old": diff.old_value,  # omitted if old_value is MISSING
+                "new": diff.new_value,  # omitted if new_value is MISSING
+                "patch": true if patched else false
+                "reason": "...",  # omitted if reason for change is unknown
+            },
+            ...
+        ]
+    }
+    ```
+    """
+    diffs = [diff_to_json(d) for d in sorted(case.diffs, key=lambda d: d.path)]
+    data = {"case_id": case.case_id, "diffs": diffs}
+    return f"<diff>{escape(json.dumps(data))}</diff>"
```

Comment:
what do you think about a slightly more expressive name for this block, like `<case_migration_diff>`

Reply:
I'm not excited about changing it at this point because some of these forms already exist on prod, and I'd rather not have to check for two different names in places that extract diff data. The form XMLNS is `http://commcarehq.org/couch-to-sql/patch-case-diff`, which clearly indicates it's related to Couch-to-SQL case diffs.